nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/bootstrap.conf

```diff
@@ -66,6 +66,15 @@ java.arg.16=-Djavax.security.auth.useSubjectCredsOnly=true
 # Please see https://zookeeper.apache.org/doc/current/zookeeperAdmin.html#sc_adminserver_config for configuration options.
 java.arg.17=-Dzookeeper.admin.enableServer=false
 
+# The following options configure a Java Agent to handle native library loading.
+# It is needed when a custom jar (eg. JDBC driver) has been configured on a component in the flow and this custom jar depends on a native library
+# and tries to load it by its absolute path (java.lang.System.load(String filename) method call).
+# Use this Java Agent only if you get "Native Library ... already loaded in another classloader" errors otherwise!
+#java.arg.18=-javaagent:./lib/aspectjweaver-${aspectj.version}.jar
```

Comment:
Do we have any concerns about aspectjweaver being directly in the lib directory which puts in on the classpath of every single NAR? Wondering if it should be separated like `lib/aspectj/` which would not be added to NiFi's normal classpath, but I don't know enough to say if it still works correctly then.

Reply:
The agent does not require `aspectjweaver.jar` on the classpath so `lib/aspectj` directory could work.
If there are concerns about having this jar on the system classpath, I can try out that solution too.

Reply:
I think if we can do the `lib/aspectj` approach then that is the best option, anything we can do to avoid introducing additional dependencies to the system classpath is good.

Reply:
I double checked it and unfortunately the aspect class in `nifi-nar-utils.jar` depends on `aspectjrt.jar`.
As `nifi-nar-utils.jar` is located in the lib folder and loaded by the system classloader, `aspectjrt.jar` must be on the system classpath too.
And I did not want to separate the 2 AspectJ jars so I put `aspectjweaver.jar` to the lib folder.

Options:
1. leave it as it is now (both `aspectjrt.jar` and `aspectjweaver.jar` in the `lib` folder)
2. leave `aspectjrt.jar` in `lib` and move `aspectjweaver.jar` to `lib/aspectj`
3. find another place for the aspect class and move its `aspectjrt.jar` dependency there, move `aspectjweaver.jar` to `lib/aspectj`

Option 2 seems to me a bit weird structure and one of the AspectJ jars is still on the system classpath.
Option 3 could work but I would keep the native library loading stuff (`AbstractNativeLibHandlingClassLoader` and `LoadNativeLibAspect`) in the same module.

So I would still vote for option 1.
These jars have no effect if they are not referenced in our code / the agent is not turned on.

Reply:
Well it does have an effect... If the jars are in lib then they are on the classpath of every NAR, so if someone makes a custom NAR that bundles a different version of aspectj, then there will be two versions on the classpath, and there is nothing the NAR can do to avoid the one in lib.

I would actually vote for option 3... create a new module like `nifi-nar-aspect-utils` or whatever we want to call it, and put the dependency there along with `LoadNativeLibAspect`, then `lib/aspectj` can contain `nifi-nar-aspect-utils` jar and the aspectj jars. Would that work?

Reply:
It has been revealed that `aspectjrt` does not needed at runtime, only `aspectjweaver` (which is loaded directly by the agent, not from classpath).
So I removed `aspectjrt.ajr` from `lib` and moved `aspectjweaver.jar` to `lib/aspectj` subdirectory.
In this way, there are no AspectJ libraries on the system classpath by default (when the agent is turned off).

When the agent is turned on, it is being loaded by the system classloader (even if `aspectjweaver.jar` is not on the system classpath). So there might be collision between different versions in this case. I believe it is acceptable because only the "agent turned on" case affected.